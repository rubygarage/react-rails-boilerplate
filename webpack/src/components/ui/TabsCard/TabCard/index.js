import React, { Component } from 'react'
import PropTypes from 'prop-types'
import classNames from 'classnames'
import Icon from 'components/ui/Icon'

import styles from './tab-card.css'

export class TabCard extends Component {
  static propTypes = {
    className: PropTypes.string,
    text: PropTypes.string,
    type: PropTypes.string,
    icon: PropTypes.string,
    color: PropTypes.string,
    iconClass: PropTypes.string,
    onClick: PropTypes.func,
    tabName: PropTypes.string
  }

  handleTabSwitch = () => {
    const { tabName } = this.props
    if (tabName === 'text') { // TODO remove this when text card will be implemented
      return
    }

    this.props.onClick(tabName)
  }

  render() {
    const { className, color, text, type, icon, iconClass, tabName } = this.props

    return (
      <div
        className={
          classNames(
            { [styles.disable]: tabName === 'text' }, // TODO remove this when text card will be implemented
            styles.tab,
            styles[type],
            styles[color],
            className,
            'tab-theme'
          )
        }
        onClick={this.handleTabSwitch}
      >
        <Icon type={icon} className={iconClass} />
        <span>
          {text}
        </span>
      </div>
    )
  }
}

export default TabCard
