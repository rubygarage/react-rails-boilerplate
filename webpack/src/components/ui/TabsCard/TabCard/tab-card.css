.tab {
  color: #4a4a4a;
  font-size: 1.2rem;
  padding: 1rem;
  cursor: pointer;
  border-bottom: 0.1rem solid rgba(255, 255, 255, 0.5);
  flex-grow: 1;
  text-align: center;
  display: flex;
  align-items: center;
  justify-content: center;

  &:hover {
     > i,
     > span {
      opacity: 0.6;
    }
  }

  &.active {
    color: #9b9b9b !important;
    border: 0.1rem solid rgba(255, 255, 255, 0.5);
    border-bottom: none;
    border-radius: 0.3rem 0.3rem 0 0;
    cursor: default;

    &:hover {
      > i,
      > span {
        opacity: 1;
      }
    }
  }
}

.disable {
  cursor: not-allowed;
  opacity: 0.3;
}

.grey {
  border-bottom: 0.1rem solid #e4e4e4;

  &.active {
    border: 0.1rem solid #e4e4e4;
    border-bottom: none;
  }
}
