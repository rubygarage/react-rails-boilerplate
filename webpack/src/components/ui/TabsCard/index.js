import React from 'react'
import PropTypes from 'prop-types'
import classNames from 'classnames'

import styles from './tabs-card.css'

function TabsCard({ className, children }) {
  return (
    <div className={classNames(styles.tabs, className)}>
      {children}
    </div>
  )
}

TabsCard.propTypes = {
  className: PropTypes.string,
  children: PropTypes.array
}

export default TabsCard
