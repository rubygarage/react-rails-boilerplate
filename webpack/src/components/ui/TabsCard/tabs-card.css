.tabs {
  display: flex;
  justify-content: space-between;
  flex-shrink: 0;
}
