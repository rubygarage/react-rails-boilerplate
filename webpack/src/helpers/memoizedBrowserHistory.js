import createHistory from 'history/createBrowserHistory';

let instance = null;

export default class memoizedBrowserHistory {
  constructor() {
    if (!instance) {
      instance = createHistory();
    }

    return instance;
  }
}
