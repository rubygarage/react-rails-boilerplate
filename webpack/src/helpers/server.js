// moved the ugly environment (server/client) check to this helper
export const isServer = () => typeof window === 'undefined'
export const isClient = () => typeof window !== 'undefined'
