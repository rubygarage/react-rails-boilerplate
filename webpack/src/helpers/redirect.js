import memoizedBrowserHistory from 'helpers/memoizedBrowserHistory'

// uses browser router or server router depending on presence of response object in arguments
export const redirect = (path, res = null) => {
  if (res) {
    res.redirect(path)
    // we need to mark the response as finished to avoid double-render/double headers being set later in the stack
    // ( see src/server.js )
    res.finished = true
  } else {
    new memoizedBrowserHistory().push(path)
  }
}
