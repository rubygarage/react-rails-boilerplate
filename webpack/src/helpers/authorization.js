import { normalize } from 'normalize-json-api';
import { setTokenToStorage, deleteTokenFromStorage } from 'utils/tokens';
import { isClient } from 'helpers/server';
import ApiClient from 'utils/apiClient';

const authorizeUser = (req, resolve) => {
  const apiClient = new ApiClient().buildClient(req);

  apiClient.get('/api/v1/auth/users/session').then((response) => {
    const { entities, results } = normalize(response.data);

    if (isClient()) { setTokenToStorage(response.headers); }
    resolve({ currentUser: results, entities });
  }).catch(() => {
    deleteTokenFromStorage();
    resolve({ currentUser: {}, entities: {} });
  });
};

export default authorizeUser;
