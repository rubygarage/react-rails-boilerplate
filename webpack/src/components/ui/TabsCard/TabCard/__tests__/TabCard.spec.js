import React from 'react'
import { shallowWithIntl } from 'helpers/intl-enzyme'
import { TabCard } from 'components/ui/TabsCard/TabCard'

describe('<TabCard />', () => {
  const props = {
    color: 'grey',
    className: 'in-white',
    icon: 'icon-user-circle',
    iconClass: 'font-20',
    onClick: jest.fn(),
    text: 'some text',
    type: 'active'
  }

  it('renders tab with text', () => {
    const component = shallowWithIntl(
      <TabCard {...props} />
    )

    expect(component).toMatchSnapshot()
  })
})
