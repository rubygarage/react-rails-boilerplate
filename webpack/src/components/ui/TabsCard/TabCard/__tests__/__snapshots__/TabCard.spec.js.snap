// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TabCard /> renders tab with text 1`] = `
<div
  className="tab active grey in-white tab-theme"
  onClick={[Function]}
>
  <Icon
    className="font-20"
    type="icon-user-circle"
  />
  <span>
    some text
  </span>
</div>
`;
